client/tests/integration/asset.rs

```diff
@@ -235,3 +235,103 @@ fn client_add_asset_with_name_length_more_than_limit_should_not_commit_transacti
 
     Ok(())
 }
+
+#[test]
+fn correct_pagination_assets_after_creating_new_one() -> Result<()> {
```

Comment:
Excellent test. I would recommend breaking it up into smaller functions that just get called, but it's readable like this also.